viam-cartographer/src/carto_facade/carto_facade_test.cc

```diff
@@ -616,6 +616,28 @@ BOOST_AUTO_TEST_CASE(CartoFacade_demo) {
                    VIAM_CARTO_POINTCLOUD_MAP_EMPTY);
     }
 
+    // GetInternalState
+    // Every call to AddSensorReading after the first one (for some
+    // unknown reason) should be persisted in the the internal state.
+    int last_internal_state_response_size = 0;
+    {
+        BOOST_TEST(viam_carto_get_internal_state_response_destroy(nullptr) ==
+                   VIAM_CARTO_GET_INTERNAL_STATE_RESPONSE_INVLALID);
+        BOOST_TEST(viam_carto_get_internal_state(nullptr, nullptr) ==
+                   VIAM_CARTO_VC_INVALID);
+        BOOST_TEST(viam_carto_get_internal_state(vc, nullptr) ==
+                   VIAM_CARTO_GET_INTERNAL_STATE_RESPONSE_INVLALID);
+
+        viam_carto_get_internal_state_response isr;
+        BOOST_TEST(viam_carto_get_internal_state(vc, &isr) ==
+                   VIAM_CARTO_SUCCESS);
+        BOOST_TEST(blength(isr.internal_state) >
+                   last_internal_state_response_size);
```

Comment:
Is the initial internal state size a known and deterministic value that we can test for instead of testing > 0?

Reply:
Never mind, if doing this requires knowing the size of data structures internal to cartographer, I don't think it's necessary. Feel free to resolve

Reply:
no it is not, it is serialized protobuf which represents internal cartographer objects.

Reply:
I don't know if it is or is not deterministic but we don't really care what it is as long as it allows cartographer to pick up from where it left off last mapping / updating session.